chains/ethereum/writer_methods.go

```diff
@@ -175,7 +174,7 @@ func (w *writer) createGenericDepositProposal(m msg.Message) bool {
 
 // watchThenExecute watches for the latest block and executes once the matching finalized event is found
 func (w *writer) watchThenExecute(m msg.Message, handler common.Address, data []byte, latestBlock *big.Int) {
```

Comment:
If we exceed the blockretrylimit waiting for a proposal finalization event should we cancel the proposal with 'cancelProposal'?

Reply:
Awesome idea. This would actually be quite trivial to implement, but but hesitant as its not presently a requirement. Could you open an issue and we can address in the near future?